R/benchmarks.R

```diff
@@ -2,23 +2,52 @@
 #'
 #' @description Tests that the shapley algorithm converges.
 #' @param row.nr Index for the observation of interest.
-#' @param iterations Amount of iterations.
-#' @param shapley.iterations Amount of the iterations within the shapley
-#'   function.
-#' @return shapley value as a data.frame with col.names and their corresponding
-#'   effects.
-test.covergence = function(row.nr, iterations = 20, shapley.iterations = 1) {
-  values = rep(0, times = iterations)
-  for(i in 1:iterations) {
-    values[i] = sum(shapley(row.nr, iterations = shapley.iterations))
-  }
+#' @param convergence.iterations Amount of calls of the shapley function.
+#' @param iterations Amount of the iterations within the shapley function.
+#' @return shapley values as a data.frame or a plot
+#'
+test.convergence = function(row.nr=2, convergence.iterations = 20, iterations = 20, task = iris.task,
+                            model = train(makeLearner("classif.lda", predict.type = "prob"), iris.task),
+                            return.value = "values") {
 
-  return(values)
-}
+  assert_number(iterations, convergence.iterations, row.nr)
+  assert_choice(return.value, c("plot", "values"))
+  assert_class(model, "WrappedModel")
+
+  data = getTaskData(task)[row.nr,]
+  prediction = getPredictionData(data, model, task)
+  prediction.class = names(prediction[BBmisc::getMaxIndex(prediction)])
 
-#a = test.covergence(196, iterations = 20)
-#a.cum = cumsum(a) / seq_along(a)
-#a.sub = a
-#ggplot() +
-#  geom_point(aes(x = seq_along(a), y = a)) +
-#  geom_line(aes(x = seq_along(a), y = cumsum(a)/seq_along(a)), color = "red")
+  values = rep(0, times = convergence.iterations)
+  for(i in 1:convergence.iterations) {
+    if(getTaskType(task) == "classif"){
+      shap = shapley(row.nr, task, model, iterations)

```

Comment:
This is the same as line 28. It doesn't matter what condition we have, we always do this. So can you put that line before the `if(...)`-block into line 23 but after the `for`-loop